DirectProgramming/DPC++FPGA/Tutorials/Features/mem_channels/README.md

```diff
@@ -0,0 +1,310 @@
+
+The Intel® oneAPI DPC++ compiler allows you to avoid this area overhead by
+disabling burst-interleaving and assigning buffers to individual channels. There
+are two advantages to such configuration:
+1. A simpler global memory interconnect is built, which requires a smaller
+   amount of FPGA resources than the interconnect needed for the
+   burst-interleaving configuration.
+2. Potential improvements to the global memory bandwidth utilization due to
+   less contention at each memory channel.
+
+Burst-interleaving should only be disabled in situations where satisfactory
+load balancing can be achived by assigning buffers to individual channels.
+Otherwise, the global memory bandwidth utilization may be reduced, which will
+negatively impact the throughput of your design. 
+
+To disable burst-interleaving, you need to pass the `-Xsno-interleaving` flag
+to your `dpcpp` command. With interleaving disabled, you can now specify, using
+the `mem_channel` property, in which memory channel each buffer should be
+allocated:
+```c++
+buffer a_buf(a_vec, {property::buffer::mem_channel{1}});
+buffer b_buf(b_vec, {property::buffer::mem_channel{2}});
+```
+The channels are assumed to have IDs `1,2,...,N` where `N` is the number of
+available channels.
+
+For FPGA boards that have multiple memory types, it is possible to select which
+memory you want to disable burst-interleaving for by passing the memory type to
+the `-Xsno-interleaving` flag: `-Xsno-interleaving=<global_memory_type>`. The
+memory type is usually indicated in the board specification XML file.
+
+An important observation is that the `mem_channel` property is a runtime
+specification which means you can change the selected channel ID for each
+buffer without having to recompile the device code. In other words, when you
+change the channel IDs, you only need to recompile the host code since
+`mem_channel` has no impact on the RTL generated by the compiler.
+
+### Tutorial Design 
+The basic function performed by the tutorial kernel is an addition of 3
+vectors. When burst-interleaving is disabled, each buffer is assigned to a
+specific memory channel depending on how many channels are available. 
+
+Due to the nature of this design and the fact that all the buffers have the
+same size, we expect accesses to the different memory channels to be well
+balanced. Therefore, we can predict that disabling burst-interleaving will not
+impact the throughput of the design, and so, it is likely beneficial to disable
+interleaving to avoid the area overhead imposed by the interleaving logic.
+
+This tutorial requires compiling the source code twice: once with the
+`-Xsno-interleaving` flag and once without it. In the `CMakeLists.txt` file,
+the macro `NO_INTERLEAVING` is defined when the `-Xsno-interleaving` flag is
+passed to the `dpcpp` command. The macro controls whether the buffers are
+created with our without the `mem_channel` property.
```

Comment:
What happens if my code uses the mem_channel property, but the no-interleaving flag is not specified when compiling?  Whatever the answer, should we mention it here?

Corollary to that question, what happens if we compile with no-interleaving but don't use the mem_channel property?  Will all buffers end up in the same memory?  Or is the compiler free to choose any buffer for each memory?  In cases where you don't care about memory bandwidth at all that might be fine.

Reply:
Addressed.